scheduler.py

```diff
@@ -264,6 +368,29 @@ def register_worker_round_robin_mode(self, assignedWorkerId, workerInfo: WorkerI
         
         logging.info(f"Finished registering the worker in {self.schedulerMode} mode with {assignedWorkerId}. WorkerInfo {workerInfo}")
 
+    def register_worker_power_of_two(self, assignedWorkerId, workerInfo: WorkerInfo):
+        mapLockName = self.WorkersPowerOf2WithoutGPULock
+        mapName = self.WorkersPowerOf2WithoutGPU
+
+        listLockName = self.WorkersListPowersOf2WithoutGPULock
+        listName = self.WorkersListPowersOf2WithoutGPU
+
+        if workerInfo.isGPUEnabled is True:
+            mapLockName = self.WorkersPowerOf2WithGPULock
+            mapName = self.WorkersPowerOf2WithGPU
+            listLockName = self.WorkersListPowersOf2WithGPULock
```

Comment:
Is the list primarily used only from random access?
If yes, can we utilise just the map for that? [how to random access on dict](https://stackoverflow.com/questions/4859292/how-can-i-get-a-random-key-value-pair-from-a-dictionary)

Reply:
Yes - But the method you shared does a O(n) copy of keys to a list for every call and thus maintained a separate list

Reply:
Makes sense